public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts

```diff
@@ -0,0 +1,83 @@
+import { PluginSignatureType } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { getPluginSettings } from '../pluginSettings';
+
+type SandboxEligibilityCheckParams = {
+  isAngular?: boolean;
+  pluginId: string;
+};
+
+type SandboxEnabledCheck = (params: SandboxEligibilityCheckParams) => Promise<boolean>;
+
+/**
+ * We allow core extensions to register their own
+ * sandbox enabled checks.
+ */
+const sandboxEnabledChecks: SandboxEnabledCheck[] = [isPluginFrontendSandboxEnabled];
+
+export function addSandboxEnabledCheck(checker: SandboxEnabledCheck) {
+  sandboxEnabledChecks.push(checker);
+}
+
+export async function shouldLoadPluginInFrontendSandbox({
```

Comment:
are there tests for this?

Reply:
there are e2e tests for this. I can add some unit tests